app/main.py

```diff
@@ -1,3 +1,24 @@
+from math import ceil
+
+
 class OnlineCourse:
-    # write your code here
-    pass
+    def __init__(self, name: str, description: str, weeks: int):
+        self.name = name
+        self.description = description
+        self.weeks = weeks
+
+    def __str__(self):
+        return self
+
+    @staticmethod
+    def days_to_weeks(days: int):
+        return ceil(days / 7)
+
+    @classmethod
+    def from_dict(cls, course_dict: dict):
+        days = course_dict["days"]
+        return OnlineCourse(
+            course_dict["name"],
+            course_dict["description"],
+            OnlineCourse.days_to_weeks(days)
+        )
```

Comment:
It would be better to use `return cls(...)` here. You will learn about this case in the future videos on OOP

Reply:
I'm doing my next task and have just read about that)) i will rework it

Reply:
good) make commit)